src/networking.c

```diff
@@ -3430,7 +3445,11 @@ int checkClientOutputBufferLimits(client *c) {
 int closeClientOnOutputBufferLimitReached(client *c, int async) {
     if (!c->conn) return 0; /* It is unsafe to free fake clients. */
     serverAssert(c->reply_bytes < SIZE_MAX-(1024*64));
-    if (c->reply_bytes == 0 || c->flags & CLIENT_CLOSE_ASAP) return 0;
+    /* Since replica clients use global replication buffer and don't use
+     * private reply list, its reply_bytes always is 0, so we should not
+     * judge it for replica clients. */
```

Comment:
i think it's just enough to mention why we skip part of the check
```suggestion
    /* Note that c->reply_bytes is irrelevant for replica clients (they use the global repl buffers). */
```